stacker_blueprints/eks.py

```diff
@@ -0,0 +1,427 @@
+        "sed -i s,DNS_CLUSTER_IP,$DNS_CLUSTER_IP,g  /etc/systemd/system/kubelet.service", "\n",
+        "sed -i s,CERTIFICATE_AUTHORITY_FILE,$CA_CERTIFICATE_FILE_PATH,g /var/lib/kubelet/kubeconfig", "\n",
+        "sed -i s,CLIENT_CA_FILE,$CA_CERTIFICATE_FILE_PATH,g  /etc/systemd/system/kubelet.service", "\n",
+        "systemctl daemon-reload", "\n",
+        "systemctl restart kubelet", "\n",
+        "/opt/aws/bin/cfn-signal -e $? ",
+        "         --stack ", Ref("AWS::StackName"),
+        "         --resource NodeGroup ",
+        "         --region ", Ref("AWS::Region"), "\n"
+    ]
+    return Base64(Join("", launch_config_userdata))
+
+
+class Workers(Blueprint):
+    VARIABLES = {
+        "ClusterName": {
+            "type": str,
+            "description": "The name of the cluster for workers to join."
+        },
+        "WorkerSecurityGroupId": {
+            "type": str,
+            "description": "The security group ID which will contain worker "
+                           "nodes."
+        },
+        "MinInstanceCount": {
+            "type": int,
+            "description": "The minimum number of worker nodes for the worker "
+                           "AutoScalingGroup.",
+            "default": 1,
+        },
+        "MaxInstanceCount": {
+            "type": int,
+            "description": "The maximum number of worker nodes for the worker "
+                           "AutoScalingGroup.",
+            "default": 3,
+        },
+        "DesiredInstanceCount": {
+            "type": int,
+            "description": "The desired number of worker nodes for the worker "
+                           "AutoScalingGroup. Defaults to minimum.",
+            "default": -1,
+        },
+        "WorkerSubnets": {
+            "type": str,
+            "description": "A list of subnet ID's where workers will be "
+                           "launched."
+        },
+        "ImageId": {
+            "type": str,
+            "description": "Worker node AMI. You need to use one of the AWS "
+                           "provided EKS worker AMI's."
+        },
+        "InstanceType": {
+            "type": str,
+            "description": "Instance type for workers.",
+            "default": "t2.small",
+        },
+        "KeyName": {
+            "type": str,
+            "description": "Existing SSH key name for worker access."
+        },
+        "RootVolumeSize": {
+            "type": int,
+            "description": "Root volume size in GB.",
+            "default": 20,
+        },
+        "RootVolumeDevice": {
+            "type": str,
+            "description": "The block device name for the root volume. This "
+                           "will depend on instance type and AMI",
+            "default": "/dev/sda1"
+        }
+    }
+
+    def create_node_instance_role(self):
+        t = self.template
+
+        # The user data below relies on this map being present.
```

Comment:
This seems out of place here, since it's not used in this method directly - though maybe not necessary per previous comment.

Reply:
I don't understand this comment. t is used to create an Output below